### PR TITLE
fix(GH-3457): Use client/secret auth for Keycloak Admin Client

### PR DESCRIPTION
**Pull request by igdianov**
- [x] Removed Activiti Keycloak Admin Client user from realm
- [x] Added  Activiti  Keycloak Admin Client with Service account user to realm
- [x] Added deployment support for Activiti Keycloak Admin Client with Client Credentials using Helm values:

```yaml
global:
  keycloak:
    clientId: activiti-keycloak
    # global.keycloak.clientSecret -- Configure Keycloak Client Secret. Required if useExistingClientSecret set to false
    clientSecret: ""
    # global.keycloak.clientSecretName -- Configure Keycloak Client Secret Name. Secret will be created if `useExistingClientSecret` set to false
    clientSecretName: activiti-keycloak-secret
    # global.keycloak.useExistingClientSecret -- Configure to use existing Keycloak clientId and clientSecret from Kubernetes Secret
    useExistingClientSecret: false
```

To create default Activiti Keycloak Client Kubernetes secret with Helm, use the following values:

```yaml
global:
  keycloak:
    clientSecret: changeit
```

To use existing Activiti Keycloak Client Kubernetes secret use the following Helm values:

```bash
kubectl create secret generic activiti-keycloak-client \
   --from-literal=clientId=activiti-keycloak \
   --from-literal=clientSecret=changeit
```

```yaml
global:
  keycloak:
    clientSecretName: activiti-keycloak-client
    useExistingClientSecret: true
```

Under the hood, the deployment will add the following environment variables to container specs:

```yaml
specs:
  containers:
    - name: runtime-bundle
      env:
        - name: ACTIVITI_KEYCLOAK_CLIENT_ID
          valueFrom:
            secretKeyRef:
              key: clientId
              name: activiti-keycloak-secret
        - name: ACTIVITI_KEYCLOAK_CLIENT_SECRET
          valueFrom:
            secretKeyRef:
              key: clientSecret
              name: activiti-keycloak-secret
```

Part of https://github.com/Activiti/Activiti/issues/3457

### CODE REVIEW

**Review by erdemedeiros — APPROVED** [review]
LGTM!
@igdianov, could you please, add the documentation about the usage you added in the PR description to the main README as well?

**Comment by igdianov** [comment]
@erdemedeiros Added documentation about Keycloak Client Kubernetes secret configuration

File: charts/activiti-cloud-full-example/Makefile [hunk 1]
```diff
@@ -3,13 +3,15 @@ VERSION := $(or ${VERSION}, $(shell cat ../../VERSION))
 PREVIEW_NAMESPACE_LOWERCASE := $(shell echo ${PREVIEW_NAME} | tr "[:upper:]" "[:lower:]")
 GITHUB_CHARTS_BRANCH := $(or ${GITHUB_CHARTS_BRANCH}, gh-pages)
 GITHUB_CHARTS_DIR := activiti-cloud-helm-charts
+CLIENT_SECRET := $(or ${CLIENT_SECRET}, dry-run)
```
Comment (mteodori) [inline charts/activiti-cloud-full-example/Makefile:6]: can we change to KEYCLOAK_CLIENT_SECRET to be more speaking?

File: README.md [hunk 2]
```diff
@@ -77,7 +96,7 @@ helm template --validate \
   activiti charts/activiti-cloud-full-example
 ```
 
-## Enabling partioning
+## Enabling message partitioning
```
Comment (mteodori) [inline README.md:99]: this is ok, but should have been in a separate README cleanup PR :)

File: README.md [hunk 3]
```diff
@@ -30,6 +30,13 @@ Update all dependencies:
 ```shell
 helm dependency update charts/activiti-cloud-full-example
 ```
+Create Activiti Keycloak Client Kubernetes secret in the current namespace:
```
Comment (mteodori) [inline README.md:33]: I like this but just have one concern, is it ok to do it here or would it be better to have a separate PR as I believe this is not the only sensitive value which would benefit from such a treatment and at least we should have a follow up PR that "completes" this also for other password/secrets we have

**Review by mteodori — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (erdemedeiros) [reply README.md:33]: IMO, it's fine to have it here, as it's handling new properties introduced by this PR and did not exist before. However, it makes sense to create another PR to handle other sensitive values in a similar way.

File: charts/activiti-cloud-full-example/Makefile (hunk 1 above)
Comment (erdemedeiros) [reply charts/activiti-cloud-full-example/Makefile]: Done

**Pull request merged**